imgui.cpp

```diff
@@ -3779,7 +3793,14 @@ ImGuiKeyModFlags ImGui::GetMergedKeyModFlags()
 void ImGui::NewFrame()
 {
     IM_ASSERT(GImGui != NULL && "No current context. Did you call ImGui::CreateContext() and ImGui::SetCurrentContext() ?");
+
     ImGuiContext& g = *GImGui;
+    
+    // Remove pending delete hooks before frame start.
+    // This deferred removal avoid issues of removal while iterating the hook vector
+    for (int n = g.Hooks.Size-1; n >= 0; --n)
+        if (g.Hooks[n].Type == ImGuiContextHookType_PendingRemoval)
+            g.Hooks.erase(&g.Hooks[n]);
 
```

Comment:
Why iterating backward here?

Reply:
The alternative it to not increment n, when removing an entry. Whichever you prefer.

Reply:
Yes I guess both work.
I would personally do `g.Hooks.erase(&g.Hooks[n--]);` in this small loop and iterate forward but both are ok, if you keep yours add the spacing for `-1`. Sorry to be petty :)

Reply:
I understand the love of having your project just right.